cmd/lncli/cmd_invoice.go

```diff
@@ -1,14 +1,86 @@
 package main
 
 import (
+	"encoding/base64"
 	"encoding/hex"
+	"encoding/json"
 	"fmt"
 	"strconv"
 
 	"github.com/lightningnetwork/lnd/lnrpc"
 	"github.com/urfave/cli"
 )
 
+// newHopHint creates a new lnrpc.HopHint from the provided parameters.
+func newHopHint(
+	nodeID string, chanID uint64, feeBaseMsat, feeProportionalMillionths,
+	cltvExpiryDelta uint32) (*lnrpc.HopHint, error) {
+
+	return &lnrpc.HopHint{
+		NodeId: nodeID, ChanId: chanID, FeeBaseMsat: feeBaseMsat,
+		FeeProportionalMillionths: feeProportionalMillionths,
+		CltvExpiryDelta:           cltvExpiryDelta,
+	}, nil
+}
+
+func validateAndParseRouteHints(
+	routeHints string, invoiceHints *[]*lnrpc.HopHint,
+) error {
+
+	var jsonHints []*lnrpc.HopHint
+	err := json.Unmarshal([]byte(routeHints), &jsonHints)
+	if err != nil {
+		return fmt.Errorf(
+			"error parsing route hints JSON: %w", err,
+		)
+	}
+	if len(jsonHints) == 0 {
+		return fmt.Errorf("no valid route hints found in JSON")
+	}
+	for _, jsonHint := range jsonHints {
+		if jsonHint.NodeId == "" {
+			return fmt.Errorf("node_id is missing in route hint")
+		}
+		if jsonHint.ChanId == 0 {
+			return fmt.Errorf(
+				"chan_id is missing or invalid in route hint",
+			)
```

Comment:
these checks should be in the constructor itself (and then the error return param does make sense to keep 👍 )